src/rdf4cpp/rdf/parser/IStreamQuadIterator.hpp

```diff
@@ -0,0 +1,162 @@
+#ifndef RDF4CPP_ISTREAMQUADITERATOR_HPP
+#define RDF4CPP_ISTREAMQUADITERATOR_HPP
+
+#include <iterator>
+#include <memory>
+
+#include <nonstd/expected.hpp>
+#include <rdf4cpp/rdf/Quad.hpp>
+
+namespace rdf4cpp::rdf::parser {
+
+/**
+ * An error that might occur while parsing a turtle file
+ */
+struct ParsingError {
+    enum struct Type {
+        EofReached,
+        BadSyntax,
+        BadCurie,
+    };
+
+    Type error_type;
+    uint32_t line;
+    uint32_t col;
```

Comment:
```suggestion
    uint64_t line;
    uint64_t col;
```

RDF files tend to be huge. 2^32 lines is not enough and 2^32 in a line might get hit even earlier, e.g. if a turtle file is stored in minimal form without \n.

Reply:
Good point just one problem is that serd itself uses 32bit ints 😅 (see [SerdError](https://drobilla.gitlab.io/serd/c/html/api/serd_event.html#c.SerdError)). I guess with the manual line tracking implemented currently it's not a problem but I had an idea yesterday that might make that unecessary (also faster). So I guess keep manual tracking for correctness?

Reply:
No longer a problem since I also patched that in serd